### PR TITLE
Maps geocoding changes

### PR DESCRIPTION
**Pull request by b2ddg**
Changing the geocoder endpoint here. Also adjusting the relevancy filter to use returned relevance information. 

@nilnilnil @tommytommytommy @brianrisk

### CODE REVIEW

File: lib/DDG/Spice/Maps/Maps.pm [hunk 1]
```diff
@@ -4,9 +4,10 @@ package DDG::Spice::Maps::Maps;
 use strict;
 use DDG::Spice;
 
-spice to => 'http://open.mapquestapi.com/nominatim/v1/search?format=json&json_callback={{callback}}&addressdetails=1&polygon=1&bounded=0&q=$1';
+spice to => 'http://api.mapbox.com/v4/geocode/mapbox.places/$1.json?access_token={{ENV{DDG_SPICE_MAPBOX_KEY}}}&callback={{callback}}';
 spice is_cached => 0;
 spice proxy_cache_valid => "418 1d";
+spice wrap_jsonp_callback => 1;
```
Comment (moollaza) [inline lib/DDG/Spice/Maps/Maps.pm]: This isn't necessary if the API has `callback` param. It's only for when the API doesn't let you specify `&callback={{callback}}` so we can tell NGINX to wrap the JSON response.

Comment (b2ddg) [inline lib/DDG/Spice/Maps/Maps.pm]: Thanks @moollaza, I removed the callback param from the request url.

File: lib/DDG/Spice/Maps/Maps.pm [hunk 2]
```diff
@@ -4,9 +4,10 @@ package DDG::Spice::Maps::Maps;
 use strict;
 use DDG::Spice;
 
-spice to => 'http://open.mapquestapi.com/nominatim/v1/search?format=json&json_callback={{callback}}&addressdetails=1&polygon=1&bounded=0&q=$1';
+spice to => 'http://api.mapbox.com/v4/geocode/mapbox.places/$1.json?access_token={{ENV{DDG_SPICE_MAPBOX_KEY}}}';
```
Comment (nilnilnil) [inline lib/DDG/Spice/Maps/Maps.pm]: They don't support callbacks?

Comment (nilnilnil) [inline lib/DDG/Spice/Maps/Maps.pm]: Looking more at this package, it's totally broken.

File: share/spice/maps/maps/maps_maps.js [hunk 3]
```diff
@@ -1,15 +1,17 @@
 DDG.require('maps',function(){
     ddg_spice_maps_maps = function(response) {
         
-        if (!response || !response.length) { return Spice.failed('maps'); }
+        if (!response) { return Spice.failed('maps'); }
```
Comment (nilnilnil) [inline share/spice/maps/maps/maps_maps.js]: What about looking for the rest of the stuff?

```javascript
if (!response || !response.features || !response.features.length) { return Spice.failed('maps'); }
```

File: share/spice/maps/maps/maps_maps.js [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 
         // OSM sends back a bunch of places, just want the first one for now
```
Comment (nilnilnil) [inline share/spice/maps/maps/maps_maps.js]: Update the comment.

File: share/spice/maps/maps/maps_maps.js [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
-        response = [response[0]]; 
+        response = response.features[0]; 
+
+        response.address=response.place_name;
```
Comment (nilnilnil) [inline share/spice/maps/maps/maps_maps.js]: Need to do more validation here that this exists.

File: share/spice/maps/maps/maps_maps.js [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
                                 
-        if (!DDG.isRelevant(response[0].display_name.toLowerCase())) { 
+        if (response.relevance < 0.9) {
```
Comment (nilnilnil) [inline share/spice/maps/maps/maps_maps.js]: How did you determine this value?

**Comment by nilnilnil** [comment]
I left some comments. I'd prefer @bsstoner lead from here though.

File: lib/DDG/Spice/Maps/Maps.pm [hunk 7]
```diff
@@ -4,9 +4,10 @@ package DDG::Spice::Maps::Maps;
...
 spice is_cached => 0;
 spice proxy_cache_valid => "418 1d";
```
Comment (nilnilnil) [inline lib/DDG/Spice/Maps/Maps.pm]: Why can't this be cached? Perhaps questions for another day.

**Comment by tommytommytommy** [comment]
:+1: Needs to be merged and released together with internal PR 4761.

**Comment by nilnilnil** [comment]
there is a caching error; but I can't fix until it's merged -- @b2ddg is here now

**Pull request merged**